### DISCUSSION TITLE
HyperHDR using Snapcast as audio capture device

### DISCUSSION THREAD

**Discussion by peetipablo** [awawa-dev/HyperHDR#188] [Q&A]
Hey there!  I have HyperHDR working after migrating from Hyperion-NG.  The main reason I wanted to migrate to HyperHDR is the music visualizer support and the better performance.

I have HyperHDR running on a raspberry pi 4 that is powering my TV ambilight.  I also have this rpi4 running a client for snapcast multi-room audio ([https://github.com/badaix/snapcast](url)).  

I am wondering if anyone has been able to get snapcast working as the audio in to HyperHDR?  I'm having no luck so far; I've tried setting the sound capture device as the audio output of the Pi, but I don't think that's right.  HyperHDR doesn't detect any audio while music is outputting to the selected device:

> 2022-01-08T23:23:40.516Z [EFFECTENGINE0] Run effect "Music: equalizer test (turn on video preview)" on channel 1
> 2022-01-08T23:23:40.516Z [EFFECTENGINE0] (EffectEngine.cpp:152) Start the effect: name [Music: equalizer test (turn on video preview)], smoothCfg [30]
> 2022-01-08T23:23:40.516Z [MUXER0] Register new input 'Web Configuration@::ffff:192.168.20.30/EFFECT' with priority 1 as inactive
> 2022-01-08T23:23:40.517Z [HYPERHDR] Sound device is starting
> 2022-01-08T23:23:40.517Z [HYPERHDR] Opening device: plughw:CARD=Device,DEV=0
> 2022-01-08T23:23:40.520Z [HYPERHDR] Sound period size = 2048
> 2022-01-08T23:23:40.520Z [HYPERHDR] Sound buffer size = 4096
> 2022-01-08T23:23:40.526Z [HYPERHDR] Sound device has started
> 2022-01-08T23:23:40.526Z [EFFECT0(Music:...)] Begin playing the effect with priority: 1
> 2022-01-08T23:23:40.626Z [MUXER0] Priority 1 is now active
> 2022-01-08T23:23:40.626Z [MUXER0] Set visible priority to 1
> 2022-01-08T23:23:40.626Z [SMOOTHING0] Clearing queued colors
> 2022-01-08T23:23:40.627Z [SMOOTHING0] Selecting config (30) => type: Linear, dirMode: true, pause: false, settlingTime: 200ms, interval: 40ms (25Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0
> 2022-01-08T23:23:40.627Z [HYPERHDR0] New priority[1], previous [255]
> 2022-01-08T23:23:40.627Z [HYPERHDR0] New source available -> switch LED-Device on
> 2022-01-08T23:23:40.628Z [IMAGETOLED0] (ImageProcessor.cpp:180) set hard led mapping to multicolor_mean
> 2022-01-08T23:23:40.629Z [IMAGETOLED0] Total index number is: 1068. Sparse processing: disabled, image size: 80 x 45, area number: 267

**Comment by awawa-dev** [comment]
Hi
That log that last just ~0.12sec doesn't tell me anything unfortunately. From other similar stories the problem usually happens when wrong audio device is selected in HyperHDR, in the software emulation of sound device that is source of the sound but it can't provide it for selected ALSA format or in the system mixer.

**Comment by peetipablo** [comment]
Thanks for the response.  It's definitely possible that as you mentioned I have the wrong audio device selected, but I've spent an entire day playing with the configuration and selecting different audio sources.  I triple checked that audio was playing from my speakers, and confirmed the same device was selected as both the audio output and hyperHDR  but no luck.

I wonder if anyone else has had luck with this setup? Snapserver is a pretty popular homebrew multi room audio software so I can't imagine I'm the first person to try this.
